### PR TITLE
MKL DNN: Softmax dim fix: Supporting 1D to 5D tensors, address the regression of CI

### PR DESCRIPTION
**Pull request by ashraf-bhuiyan**
Softmax test failed because recently some tests uses 3D tensors as input. Before, MKL softmax only supported 2D tensors. Now, in this PR, we are supporting 1D, 2D, 3D, 4D and 5D tensors as input to Softmax. 
Thus, this PR fixes the CI regression.

### CODE REVIEW

**Comment by agramesh1** [comment]
@tatianashp

File: tensorflow/core/kernels/mkl_softmax_op.cc [hunk 1]
```diff
@@ -62,7 +63,26 @@ class MklSoftmaxOp : public OpKernel {
                               : src_tensor.shape();
       auto src_dims = TFShapeToMklDnnDims(src_tf_shape);
       auto output_dims = src_dims;
-
+      memory::format layout_type;
+      switch (input_dims) {
+        case 1:
+          layout_type = memory::format::x;
```
Comment (tatianashp) [inline tensorflow/core/kernels/mkl_softmax_op.cc]: Could you please add comments for the 5 cases here explaining what these comments are or giving their full names.

Thanks!

Comment (tatianashp) [reply tensorflow/core/kernels/mkl_softmax_op.cc]: Thank you for fixing the regression test failure! Looks great. Just one small request.

**Review by tatianashp — CHANGES_REQUESTED** [review]

File: tensorflow/core/kernels/mkl_softmax_op.cc (hunk 1 above)
Comment (ashraf-bhuiyan) [reply tensorflow/core/kernels/mkl_softmax_op.cc]: I have added some comments for the data format. Please let me know if this is ok.

**Review by tatianashp — APPROVED** [review]
Thanks for the comment. Approved.
